ginga/AstroImage.py

```diff
@@ -38,8 +38,7 @@ def set_ioClass(cls, klass):
         cls.ioClass = klass
 
     def __init__(self, data_np=None, metadata=None, logger=None,
-                 name=None, wcsclass=None, ioclass=None,
-                 inherit_primary_header=False, save_primary_header=True):
```

Comment:
Should keep the old signature during deprecation phase. Same comment for all similar occurrences.

Reply:
The problem is that if we leave the kwargs in we don't know whether someone passed the parameter or not.  By using the **kwargs notation we can issue the warning only if they actually passed the parameter.

I'm assuming we don't want to raise a warning just by instantiating the class, if the parameter was never passed.

Reply:
I wonder if one of the "deprecated" decorators would work for you: https://docs.astropy.org/en/latest/utils/index.html#module-astropy.utils.decorators

Reply:
Hmm, documentation says "The decorator assumes that the argument with the old_name was removed from the function signature and the new_name replaced it at the same position in the signature."

But both kwargs are being removed completely and not replaced.

I think maybe for this particular PR the deprecation decorators are not necessary. But thanks for pointing them out, I will definitely use them if renaming a class method or attribute in the future!

Reply:
I think if `new_name` is `None`, that would satisfy your situation but here you are deprecating two, so maybe applying that decorator twice is overkill, I agree.